### PR TITLE
PublicDashboards: refactor service

### PR DESCRIPTION
**Pull request by evictorero**
**What this PR does / why we need it**:
* Removes `dashboard_query.go` files and package  and moves that logic into the service
* Splits `service` into `service.go` and `query.go` with tests into `service_test.go` and `query_test.go`
* Unexports methods not used outside the package
* Renames methods and functions for better legibility

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-partnerships-team/issues/436

### CODE REVIEW

**Review by owensmallwood — APPROVED** [review]
Looks good 👍

File: pkg/services/publicdashboards/api/middleware.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func RequiresExistingAccessToken(publicDashboardService publicdashboards.Service
 		}
 
 		// Check that the access token references an enabled public dashboard
-		exists, err := publicDashboardService.AccessTokenExists(c.Req.Context(), accessToken)
+		exists, err := publicDashboardService.EnabledPublicDashboardExistsByAccessToken(c.Req.Context(), accessToken)
```
Comment (jalevin) [inline pkg/services/publicdashboards/api/middleware.go:53]: Yes! Thank you for tackling this.

What do you think about `PublicDashboardEnabledExistsByAccessToken` ? 

@owensmallwood thoughts?

**Review by jalevin — APPROVED** [review]
Looks good- one minor change on naming if we think necessary. Current name does not quite match the interface methods.

Also, it looks like we deleted 1679 lines and added 1605. Did we miss anything? I would expect this to be 1:1 code change

File: pkg/services/publicdashboards/api/middleware.go (hunk 1 above)
Comment (evictorero) [reply pkg/services/publicdashboards/api/middleware.go:53]: I can change it but in that case I will change `EnabledPublicDashboardExistsByDashboardUid ` too.

Comment (jalevin) [reply pkg/services/publicdashboards/api/middleware.go:53]: Works for me!

**Comment by evictorero** [comment]
> e we deleted 1679 lines

That's a good point but I don't expect the same numbers because with code reorg there are duplicated imports, also I removed a func not being used and it test `HasExpressionQuery`.

**Pull request merged**